VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,184 @@
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, creationDate),
+                answerPrefix + " " + relativeTime + " by author (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, creationDate),
+                answerPrefix + " " + relativeTime + " (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, creationDate),
+                answerPrefix + " " + relativeTime + " by author");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, label, null),
+                answerPrefix + " " + "by author (label)");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, creationDate),
+                answerPrefix + " " + relativeTime);
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(author, null, null),
+                answerPrefix + " " + "by author");
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, label, null),
+                answerPrefix + " " + "(label)");
+
+        // Empty Case
+        applyAssertions(DiscussionTextUtils.AuthorAttributionLabel.ANSWER,
+                new StubAuthorData(null, null, null),
+                null);
+    }
+
+    private void applyAssertions(DiscussionTextUtils.AuthorAttributionLabel type, IAuthorData input,
+                                 String expectedOutput) {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView, type, input, now,
+                dummyClickListener);
+        if (expectedOutput == null) {
+            assertTrue(dummyTextView.getVisibility() == View.GONE);
+        } else {
+            String output = dummyTextView.getText().toString();
+            assertEquals(expectedOutput, output);
+        }
+    }
+
+    @Test
+    public void testAuthorIsTappable() {
+        DiscussionTextUtils.setAuthorAttributionText(dummyTextView,
+                DiscussionTextUtils.AuthorAttributionLabel.POST,
+                new StubAuthorData("testAuthor", "testLabel", new Date()),
+                dummyClickListener);
+        ClickableSpan[] spans = getClickableSpans();
+        if (config.isUserProfilesEnabled()) {
+            assertTrue(spans.length > 0);
+            spans[0].onClick(dummyTextView);
+            Mockito.verify(dummyClickListener).run();
+        } else {
+            assertTrue(spans.length == 0);
+        }
+    }
+
+    private ClickableSpan[] getClickableSpans() {
+        SpannableString spannableString = (SpannableString) dummyTextView.getText();
+        return spannableString.getSpans(0, spannableString.length(), ClickableSpan.class);
```

Comment:
This method is only being called from one place, so this code could just be included there.